MOB-1647 Check inactive lock icon

d4r1091 opened the pull request:
[MOB-1647](https://ecosia.atlassian.net/browse/MOB-1647)

## Context

We have realized that a rework of the locker icon telling us the state of things like connection and tracking protection needed a UI refresh.

## Approach

With the design team's help, we realized that simplifying that icon would address the intended UI refresh, bringing a more consistent approach to the value we wanted to deliver.

We have simplified the states (previously 3).
Here is the full breakdown:

### Previously

- tracking protection enabled + website utilizing HTTPS
- tracking protection disabled + website utilizing HTTPS
- tracking protection enabled + website utilizing HTTP
- tracking protection disabled + website utilizing HTTP

#### Now

Regardless of the tracking protection state, we differentiate two states only:

- website utilizing HTTPS -> showing the locker locked icon green highlighted
- website utilizing HTTP -> showing the alarm signal, tinted with `warning` color

## Other

This PR also addresses a bit of refactoring and icons cleanup.

kimar commented on Client/Frontend/Browser/EnhancedTrackingProtection/EnhancedTrackingProtectionVM.swift:
```diff
@@ -31,13 +31,12 @@ class EnhancedTrackingProtectionMenuVM {
 
     var connectionSecure: Bool {
-        return tab.webView?.hasOnlySecureContent ?? false
+        return tab.webView?.url?.isHTTPS == true
```
👏  Very nice!

I wonder if we'd even go for a three-stage icon:

Insecure, Mixed and `hasOnlySecureContent`...

Actually, that said I'm not even sure if the WebView loads insecure content if the root site is secure.

d4r1091 replied:
Yeah I wasn't 💯 sure about it, but was I 💯 sure of is the fact that `hasOnlySecureContent` isn't actually the best check as some underlying content (like images or resources) may not be under `https` or secure connetions

kimar commented on Client/Frontend/Toolbar+URLBar/ConnectionButton.swift:
```diff
@@ -0,0 +1,44 @@
+        updateButtonImageAccordingToStatus(status)
+    }
+    
+    func evaluateNeedingVisbilityForURLScheme(urlScheme: String?) -> Bool {
```
💡  There's. a typo in visibility: evaluateNeedingVisbilityForURLScheme -> evaluateNeedingVis**I**bilityForURLScheme

❓   Maybe get rid of one of the `urlScheme`? e.g.

`func evaluateNeedingVisbilityForURLScheme(_ urlScheme: String?) -> Bool {`

or

`func evaluateNeedingVisbility(for urlScheme: String?) -> Bool {`

d4r1091 replied:
How could I not notice it 🤦 . Thanks a lot for that @kimar! I'll go with the 1st one as that was the original intention

kimar reviewed (approved):
LGTM 👍  Just some minor comments.

lucaschifino commented on Client/Frontend/Toolbar+URLBar/LockButton.swift:
I've seen the approach of just removing from target instead of deleting the file if it is no longer user to make our lives easier when updating from FF in the future.

I don't think in this case it would be a problem (also have never experienced the update before to know), but wonder if that is indeed common practice might be good to do it. Do you know anything about that?

d4r1091 replied:
Agree on not removing files but only removing them from targets; that's the correct approach to having fewer issues for future upgrades.
While I'm 💯 onboarded when it comes to `.swift` files, not sure about resources (like those `xcassets` I've explicitly removed from the repo).
However, I don't want to bring a diversion in the standard approach we've been having so far.
@ecotopian WDYT?

ecotopian replied:
There's no real or right or wrong here, and there's no such thing as a real standard approach as every case is different. You both brought the most important arguments already. For swift-files I'd encourage to keep the class and just remove reference. In case of assets I'd advocate
 for completely removing them. Even if, it will be a conflict that will be easy to solve.

lucaschifino commented on Client/Frontend/Toolbar+URLBar/ConnectionButton.swift:
On the same line of thought as the removed file (keeping code separate so it is easier to update from FF), doesn't it make sense to add files fully owned by us to the `Ecosia` folder? That would apply to the other added files on this PR as well.

d4r1091 replied:
I see the point, yeah and I see value in your suggestion 👍 . Happy to review it so that we can have an easier version update.

lucaschifino commented on Client/Frontend/Toolbar+URLBar/ConnectionStatusImage+WebSiteConnectionTypeStatus.swift:
```diff
@@ -0,0 +1,18 @@
+
+import Shared
+
+extension ConnectionStatusImage {
```
I think personally I would have kept `getForStatus` as part of the other file and not a separate extension since it is an integral part of it and only ever used with that (the other variables could even be private).

Even so this is not major and we do not have any defined styling guidelines, so I'm also ok with leaving it like this.

d4r1091 replied:
Thanks for that!
My thought was that `ConnectionStatusImage` and `WebsiteConnectionTypeStatus`, despite being utilized to solve one scenario, they are conceptually two different things.
`ConnectionStatusImage` strictly belongs to the UI domain while `WebsiteConnectionTypeStatus` might be utilized for non-UI specific business logic.

lucaschifino commented on Client/Frontend/Toolbar+URLBar/ConnectionButton.swift:
```diff
@@ -0,0 +1,44 @@
+final class ConnectionButton: UIButton {
+        
+    // MARK: - Initializers
+
```
Also minor, but would suggest removing the space below this and the next mark.

```suggestion

```

lucaschifino commented on Client/Frontend/Browser/EnhancedTrackingProtection/EnhancedTrackingProtectionVM.swift:
```diff
@@ -31,13 +31,12 @@ class EnhancedTrackingProtectionMenuVM {
-        let insecureImageString = "lock_blocked"
-        let image = connectionSecure ? UIImage(named: "secureLock")! : UIImage(imageLiteralResourceName: insecureImageString)
-        return image
+        let connectionStatus: WebsiteConnectionTypeStatus = connectionSecure ? .secure : .unsecure
```
Also wondering if we should add the `//Ecosia: ...` comment when changing an FF file, not sure if that is relevant for every change and actually helps when merging though 🤔

d4r1091 replied:
I can add yeah 👍

lucaschifino reviewed (commented):
Nice work 🙌 Just added some comments mostly with regards to separating our code from FF's for future merges

ecotopian reviewed (commented):
Great work! 🙌  Just a tiny request to move new classes to the Ecosia folder. This will ease re-adding all the Ecosia-files to the project after next upgrade.

d4r1091 commented:
I addressed the comments boyz! Mostly adding comments and moving folders as suggested 💪 .
The `xcassets` have been left as "removed" and with regards to the `LockerButton.swift`; that is not present in the latest FF codebase.

@lucaschifino @ecotopian

lucaschifino commented on Client/Frontend/Toolbar+URLBar/ConnectionButton.swift:
Any reason not to also move this class inside the `Ecosia` folder? Isn't it a new one?

lucaschifino commented on Shared/WebsiteConnectionStatus.swift:
Same for this one, isn't it a new class and wouldn't it make sense to move to the `Ecosia` folder?

lucaschifino reviewed (approved):
Looking good! 👏 

Added comments to two files that I believe are still new ones and if so could be moved into the `Ecosia` folder as well, but no blockers from my side 🟢

d4r1091 commented:
Thanks @lucaschifino ! Updated 💪

lucaschifino reviewed (approved):
LGTM 👏
